matlab/CARFAC_AGC_Step.m

```diff
@@ -22,28 +22,47 @@
 % one time step of the AGC state update; maybe decimates internally.
 % AGC input is detects; detect_scale is already in the coefficients.
 
-if coeffs.non_decimating
-  % 2025 new decimation-free structure, intended to be GPU friendly.
-  % state is a column per stage, to be updated in parallel.
-  % First the time-update step with detects as input;
-  % new state is a linear function of old state and inputs, and we allow
-  % a sample of delay per stage, not using each new stage state until the
-  % the next time step, to make it parallelize efficiently and to make
-  % the delay a bit closer to what happens in the decimating version.
-  state.AGC_memory = ...  % Broadcasting gain rows across channels
-    coeffs.feedback_gains .* state.AGC_memory + ...
-    coeffs.input_gains .* detects + ...
-    coeffs.next_stage_gains .* ...
+if coeffs.simpler_decimating
+  % Simpler way:  Just use one decimation factor, which can be 1 or
+  % any positive integer, and update all stages in parallel; no more
+  % recursion and no more special-casing for non-decimating.
```

Comment:
For GPU friendliness, we might want to special case `coeffs.decimation(1) == 1`... but this could probably be a JAX-only change. I'm ok with leaving the MATLAB like this.